src/main/java/com/github/mtsochacki/adventofcode/Day18.java

```diff
@@ -0,0 +1,183 @@
+package com.github.mtsochacki.adventofcode;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Day18 implements Day {
+    @Data
+    @AllArgsConstructor
+    private static class RegularNumber {
+        private int level;
+        private int value;
+
+        public RegularNumber(RegularNumber regularNumber) {
+            this.level = regularNumber.level;
+            this.value = regularNumber.value;
+        }
+
+        public void incrementLevel() {
+            this.level++;
+        }
+
+        public void decrementLevel() {
+            this.level--;
+        }
+
+        public void increaseValue(int value) {
+            this.value += value;
+        }
+    }
+
+    @Override
+    public String part1(List<String> input) {
+        Integer slugMagnitude = input.stream()
+                .map(this::transformIntoSnailfishNumber)
+                .reduce(this::reduceSnailfishNumber)
+                .map(this::calculateMagnitude)
+                .orElseThrow(() -> new RuntimeException("Error processing SnailFish"));
+        return String.valueOf(slugMagnitude);
+    }
+
+    @Override
+    public String part2(List<String> input) {
+        List<List<RegularNumber>> snailfishNumbers = input.stream()
+                .map(this::transformIntoSnailfishNumber)
+                .toList();
+        int max = findHighestMagnitude(snailfishNumbers);
+        return String.valueOf(max);
+    }
+
+    private int findHighestMagnitude(List<List<RegularNumber>> snailfishNumbers) {
+        int max = 0;
+        for (int i = 0; i < snailfishNumbers.size(); i++) {
+            for (int j = 0; j < snailfishNumbers.size(); j++) {
+                if (i == j) {
+                    continue;
+                }
+                List<RegularNumber> firstSlug = copySnailfishNumber(snailfishNumbers.get(i));
+                List<RegularNumber> copyFirstSlug = copySnailfishNumber(snailfishNumbers.get(i));
+                List<RegularNumber> secondSlug = copySnailfishNumber(snailfishNumbers.get(j));
+                List<RegularNumber> copySecondSlug = copySnailfishNumber(snailfishNumbers.get(j));
+                List<RegularNumber> firstResult = reduceSnailfishNumber(firstSlug, secondSlug);
+                List<RegularNumber> secondResult = reduceSnailfishNumber(copySecondSlug, copyFirstSlug);
+                int newMax = Math.max(calculateMagnitude(firstResult), calculateMagnitude(secondResult));
+                max = Math.max(newMax, max);
+            }
+        }
+        return max;
+    }
+
+    private List<RegularNumber> copySnailfishNumber(List<RegularNumber> snailfishNumber) {
+        List<RegularNumber> newSnailfishNumber = new ArrayList<>();
```

Comment:
This may be me not knowing Java, but why not just `= new ArrayList<>(snailfishNumber)`? Internet says that it also copies

Reply:
I need actual separate copies of the objects on the lists. My explode and split actions modify original objects and it causes issues further down. `= new ArrayList<>(snailfishNumber)` makes shallow copy that gives incorrect results

Reply:
You could maybe make a case that 'explode' and 'split' should not modify their initial input... And I would probably agree with that

Reply:
I actually think `explode/split` modifying their input is okay and having dug into this a bit more you're right - you have to copy elements one by one, which is a shame.